src/Source/Source.js

```diff
@@ -1,45 +1,39 @@
 import Extent from '../Core/Geographic/Extent';
-/**
- * @typedef {Object} NetworkOptions - Options for fetching resources over the
- * network. For json or xml fetching, this object is passed as it is to fetch
- * as the init object, see [fetch documentation]{@link https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Parameters}.
- * @property {string} crossOrigin For textures, only this property is used. Its
- * value is directly assigned to the crossorigin property of html tags.
- * @property * Same properties as the init parameter of fetch
- */
 
 /**
- * @typedef {object} sourceParams
- * @property {string} protocol source's protocol (wmts, wms, wfs, file, tms, static)
- * @property {string} url Base URL of the repository or of the file(s) to load
- * @property {NetworkOptions} [networkOptions = { crossOrigin: 'anonymous' }] the base url to fetch data source
- * @property {string} [projection] data's projection
- * @property {Extent} [extent] data's extent
- * @property {Attribution} [attribution] Attribution The intellectual property rights for the source
- * @property {string} [format] data format
+ * @classdesc
+ * Sources are object containing informations on how to fetch resources, from a
+ * set source.
+ *
+ * To extend a Source, it is necessary to implement two functions: {@link
+ * Source#urlFromExtent} and {@link Source#extentInsideLimit}.
  *
+ * @property {string} url - The url of the resources that are fetched.
+ * @property {string} format - The format of the resources that are fetched.
+ * @property {Object} networkOptions - Fetch options (passed directly to
+ * <code>fetch()</code>), see {@link
+ * https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/fetch#Syntax|the
+ * syntax for more information}. By default, set to <code>{ crossOrigin:
+ * 'anonymous' }</code>.
+ * @property {string} projection - The projection of the resources.
+ * @property {string} attribution - The intellectual property rights for the
+ * resources.
+ * @property {Extent} extent - The extent of the resources.
  */
-
 class Source {
     /**
-     * Source are parameters to fetch source
-     *
-     * To extend {@link Source}, it is necessary to implement 2 functions:
-     * {@link Source#urlFromExtent} and {@link Source#extentInsideLimit}
-     * @param  {sourceParams}  source  object to set source
+     * @param {Object} source - An object that can contain all properties of a
+     * Source. Only the <code>url</code> property is mandatory.
      *
+     * @constructor
```

Comment:
Could you add `isSource` as new Source property?

Reply:
Ok